home/widgets.py

```diff
@@ -187,3 +189,53 @@ def _refresh_output(self, _=None):
             self._output.value = (
                 self.template.format(text=self.value) if self.value else ""
             )
+
+
+class CountDownWidget(ipw.VBox):
+    """A countdown w.r.t the AiiDAlab container creation time."""
+
+    TEMPLATE = "<h1 style='color: {color}'>Time remaining: <b>{countdown}</b></h1>"
+    FINAL_WARNING = "<h1 style='color: red'>AiiDAlab container shutdown imminent - please save your work!</h1>"
+
+    def __init__(self, lifetime: str, **kwargs):
+        self.reference_time = datetime.strptime(lifetime, "%H:%M:%S")
+
+        self.countdown = ipw.HTML()
+
+        super().__init__(
+            children=[
+                self.countdown,
+            ],
+            **kwargs,
+        )
+
+    def start(self):
+        Timer(1, self._update_countdown).start()
```

Comment:
Infinitely running background threads that update the UI state are unfortunately a really bad idea due to a possible memory leak described in https://github.com/aiidalab/issues/issues/13. 

I am afraid timer like this needs to be in javascript. If I am not mistaken, as long as the timer is initialized properly, it shouldn't need any information from the backend, right?